pkg/controller/dnsendpoint/dnsendpoint_controller.go

```diff
@@ -89,21 +54,90 @@ func Add(mgr manager.Manager) error {
 	if err != nil {
 		return err
 	}
+
 	if err := ctrl.Watch(&source.Kind{Type: &hivev1.DNSZone{}}, &handler.EnqueueRequestForObject{}); err != nil {
 		return err
 	}
-	return ctrl.Watch(&source.Channel{Source: nameServerChanges}, &handler.EnqueueRequestForObject{})
+
+	for _, changeNotifier := range nameServerChangeNotifiers {
+		err := ctrl.Watch(&source.Channel{Source: changeNotifier}, &handler.EnqueueRequestForObject{})
+		if err != nil {
+			log.WithField("controller", controllerName).WithError(err).Error("unable to set up watch for name server changes")
+			return err
+		}
+	}
+
+	return nil
+}
+
+func newReconciler(mgr manager.Manager, kubeClient client.Client) (*ReconcileDNSEndpoint, []chan event.GenericEvent, error) {
+	nameServerScrapers := map[string]*nameServerScraper{}
+	nameServerQueryers := map[string]nameserver.Query{}
+	nameServerChangeNotifiers := []chan event.GenericEvent{}
```

Comment:
Don't need a separate channel for each managed domain. All of the scrapers can write to the same channel.